category/traversable/basic.lean

```diff
@@ -0,0 +1,105 @@
+/-
+Copyright (c) 2017 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Type class for traversing collections
+-/
+
+import tactic.cache
+import category.applicative
+
+open function (hiding comp)
+
+universe variables u v w
+
+section applicative_transformation
+
+variables (f : Type u → Type v) [applicative f] [is_lawful_applicative f]
+variables (g : Type u → Type w) [applicative g] [is_lawful_applicative g]
+
+structure applicative_transformation : Type (max (u+1) v w) :=
+(F : ∀ {α : Type u}, f α → g α)
+(preserves_pure' : ∀ {α : Type u} (x : α), F (pure x) = pure x)
+(preserves_seq' : ∀ {α β : Type u} (x : f (α → β)) (y : f α), F (x <*> y) = F x <*> F y)
+
```

Comment:
open the `applicative_transformation` namespace, the instances and basic rules should be in it.